docs/book/features/orphan_volumes.md

```diff
@@ -0,0 +1,51 @@
+
+Orphan volumes are often created when CNS solution creates more than one FCD for a Persistent Volume in the Kubernetes cluster. This could occur when the vCenter components are slow, storage is slow, vCenter service restarts, connectivity issues between vCenter and ESXi hosts etc. Since these orphan volumes occupy space in the datastore and are not really used in Kubernetes, it's useful to identify and cleanup orphan volumes periodically.
+
+This functionality provides a set of APIs to detect and delete orphan volumes on-demand, and also provide an option to turn on automatic deletion periodically.
+
+### What qualifies as an orphan volume ?
+An FCD can qualify as an orphan volume if it meets below criteria:
+
+1) A PersistentVolume in any of the registered kubernetes clusters is not using the FCD.
+2) FCD was dynamically provisioned using vSphere CSI driver.
+3) FCD exists for more than 60 minutes(or `vso-create-time-limit-mins` value configured in `cnsmanager-config` configmap) at the time of orphan volumes detection.
+
+### Which orphan volumes are skipped from detection/deletion ?
+Some volumes, even if a kubernetes PersistentVolume doesn't map to them, will not be considered during orphan volume detection/deletion. These include:
+* FCDs created out of band (not using vSphere CSI driver).
+* Statically provisioned CNS volume whose name doesn't start with `pvc-`.
+* File volumes.
+* Orphan volumes that contain snapshots will be detected as orphans but they can not be deleted using Orphan volume delete API.
+
+### A word of caution!
+Orphan FCDs once deleted from a datastore cannot be recovered. It is therefore imperative that you [register all the Kubernetes clusters](../../../README.md#register-kubernetes-clusters-before-you-start) in vCenter with CNS Manager, so that an orphan volume is detected correctly and no crucial data loss happens when they're deleted.
+
+### APIs provided
+1. *GET /orphanvolumes* 
+
+This API takes optional parameters, datacenter & list of datastores, and returns orphan volumes for them. 
+- If datacenter is not specified, then it returns all orphan volumes in the vCenter (all datastores on all datacenters ).
+- If only datacenter is specified, then it returns orphan volumes in all datastores in the datacenter.
+- If both datacenter & list of datastores is specified, it returns orphan volumes in specified datastores on the datacenter.
+
+Detection of orphan volumes can be a time-consuming operation if there are large number of orphans. Hence it is performed asynchronously at regular intervals and the response is cached. This API returns list of orphan volumes computed in the last run, along with the next operation interval(`RetryAfterMinutes`).  
+**Note:** For newly deployed CNS manager application when orphan volumes are being computed in the background for the first time, the API may return no orphan volumes. It should then be re-tried after `RetryAfterMinutes` to get orphan volumes computed in the latest run.
```

Comment:
I did not quite understand it. Are you saying the API may not return the latest all the time and just returns from the last computed cache?

Reply:
Yes, orphan volumes are computed periodically(configurable, default 30 mins) and stored in a cache. When API is invoked, it returns orphans from the last computed cache along with a field `RetryAfterMinutes` that tells when the next cache update will happen.

Reply:
Just wondering why we went in that route, the requirement was actually to list orphan volumes at the invocation time. Is there an API that gives the current list of orphans?

Reply:
The orphan volume list API can take a long time(more than 10s of mins) if there are thousands of orphans in the system, and longer if the system is slow. That's what we discovered during scale test. Since we are anyway scanning the system periodically for any orphan volumes, it makes sense to cache them and return the response from the cache.  
 
We also don't expect new orphans to get introduced in the system frequently. If that happens, then we have a bigger problem to look into. So we should be okay to return a response from the cache. And there's no other API to give the current list of orphans.